### DISCUSSION TITLE
Connecting VSCode Desktop (Locally) to Azure Studio Compute Instance through Azure Machine Learning Extension

### DISCUSSION THREAD

**Discussion by aiplayer93** [microsoft/vscode-discussions#1245] [Extension Show and Tell]
Hello guys,, 

I have been combing the internet for this for a long long time, so your help is very very much appreciated,

I am trying to connect to an Azure Studio Compute Instance through VSCode Desktop and its associated Azure Machine Learning VSCode extension. 

**1.	Context**: While developing in Azure Studio ML Compute instance, I would need to develop in VSCode Desktop locally (VSCode Web in the Azure Studio I have is freezing and the Desktop version link in Azure Studio does not open my VSCode with administrative privileges - and for the work I do I need to develop locally with elevated privileges). 

I have tried to connect from VSCode Desktop (local VSCode) to an Azure ML Compute Instance through the Azure Machine Learning Extension (and the associated Remote) last week. It did work through VS Code Azure extension and then suddenly broke a few days later. The error message varies depending on the vs code extension that is being used. 

Usually however, the error is either that
1.  the Azure Extension is not installed on the remote or
2.  that the remote listening is not on 
3. that it failed to connect to the remote extension host server (Error: Cannot read properties of undefined (reading 'sessionId')
4. cannot read propoerties of undefined (reading 'sessionId') Source Azure Machine Learning - Remote
![image](https://github.com/microsoft/vscode-discussions/assets/113632605/e2c68cf9-8b25-43da-812e-1690388d48c6)

![image](https://github.com/microsoft/vscode-discussions/assets/113632605/521838ff-71bc-4524-83a1-5e8d0c1247a4)

![image](https://github.com/microsoft/vscode-discussions/assets/113632605/a14f51fa-d6f4-452f-9c86-427b75e3a9f7)

**2.	The question:** So I have managed to make it work before by tweaking the different version number of Azure Machine Learning and Azure Machine Learning Remote at 0.28.0 and 0.32.0 respectively (the most up to date version gave errors and I tweaked on the different versions). The VSCode version I have now is 1.90.2. Please note I have tried to create a new workspace in Azure Studio and new computes and also reinstalled VSCode to the current 1.90.2 version.

Is this a matter of version compatability and what would be the best combinations of version to make it work? What would a quick fix be for this? Should I consider the ssh route for this instead of the extension (note our compute instance is not allowed for now to have a public IP due to the vnet restrictions that are applied on the workspace).

Thanks for the help,

**Comment by xiaopangdun123** [comment]
can anyone help? I have the same problem

**Comment by gjsjohnmurray** [comment]
This isn't the right place for you to be requesting help. Please see the pinned post.

**Reply by aiplayer93** [reply]
Not sure to see the pinned post John. Where should I request for help? Thx

**Reply by isidorn** [reply]
You can also open an issue here https://github.com/microsoft/vscode-tools-for-ai

**Comment by gjsjohnmurray** [comment]
It is https://github.com/microsoft/vscode-discussions/discussions/5
